drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/client/InvocationResponseFuture.java

```diff
@@ -73,6 +79,25 @@ private InvocationResponseFuture(InvokeRequest request, ConnectionParameters con
     private synchronized void tryConnect()
     {
         try {
+            Optional<Boolean> encryptionRequired = request.getAddress().isEncryptionRequired();
```

Comment:
Sorry, I didn't notice it at the beginning. Let's move this algorithm to the `createInvocationResponseFuture` and modify the `connectionParameters` to keep the `connectionParameters` field final.